third-party-programs/Velocity-Bench/hplinpack/README.md

```diff
@@ -1,89 +1,243 @@
+
+```
+   $ source /opt/intel/oneapi/setvars.sh
+   $ dpct --version  # Intel® DPC++ Compatibility tool version
+```
+ * If want to try latest version of compatibility tool, try to install SYCLomatic by download prebuild of [SYCLomatic release](https://github.com/oneapi-src/SYCLomatic/blob/SYCLomatic/README.md#Releases) or [build from source](https://github.com/oneapi-src/SYCLomatic/blob/SYCLomatic/README.md), as follow give the steps to install prebuild version: 
+ ```
+   $ export SYCLomatic_HOME=/path/to/install/SYCLomatic
+   $ mkdir $SYCLomatic_HOME
+   $ cd $SYCLomatic_HOME
+   $ wget https://github.com/oneapi-src/SYCLomatic/releases/download/20240203/linux_release.tgz   #Change the timestamp 20240203 to latest one
+   $ tar xzvf linux_release.tgz
+   $ source setvars.sh
+   $ dpct --version #SYCLomatic version
+ ```
+ 
+ 
+For more information on configuring environment variables, see [Use the setvars Script with Linux*](https://www.intel.com/content/www/us/en/develop/documentation/oneapi-programming-guide/top/oneapi-development-environment-setup/use-the-setvars-script-with-linux-or-macos.html).
+ 
+### 2 Generate the compilation database
+``` sh
+$ cd ${hplinpack_HOME}/cuda/hpl-2.3
+$ make clean
+$ intercept-build make
+$ ls compile_commands.json  # make sure compile_commands.json is generated
+compile_commands.json
+```
+### 3 Migrate the source code and build script
+```sh
+# From the CUDA directory as root directory:
+$ cd ${hplinpack_HOME}/cuda
+$ dpct --in-root=. -p=./hpl-2.3/compile_commands.json --out-root=out --gen-build-script --cuda-include-path=/usr/local/cuda/include
+```
+Description of the options: 
+ * `--in-root`: provide input files to specify where to locate the CUDA files that needs migration.
+ * `-p`: specify compilation database to migrate the whole project.
+ * `--out-root`: designate where to generate the resulting files (default is `dpct_output`).
+ * `--gen-build-script`: generate the `Makefile.dpct` for the migrated code.
+
+Now you can see the migrated files in the `out` folder.
+### 4 Review the migrated source code and fix all `DPCT` warnings
+
+SYCLomatic and [Intel® DPC++ Compatibility Tool](https://www.intel.com/content/www/us/en/developer/tools/oneapi/dpc-compatibility-tool.html) define a list of `DPCT` warnings and embed the warning in migrated source code if need manual effort to check. All the warnings in the migrated code should be reviewed and fixed. For detail of `DPCT` warnings and corresponding fix examples, refer to [Intel® DPC++ Compatibility Tool Developer Guide and Reference](https://www.intel.com/content/www/us/en/develop/documentation/intel-dpcpp-compatibility-tool-user-guide/top/diagnostics-reference.html) or [SYCLomatic doc page](https://oneapi-src.github.io/SYCLomatic/dev_guide/diagnostics-reference.html). 
+
+Fix the warning in migrated hplinpack code:
```

Comment:
Please say someting on fix the compiler used in line 1 and Line1016,  high light the changed code to avoid reader to summary out the changeset.